### DISCUSSION TITLE
Add a button in ui.scene()

### DISCUSSION THREAD

**Discussion by Sweetharry1992** [zauberzeug/nicegui#372] [Q&A] [answered]
I want to add a button in ui.scene(). But unable to add it. I also read docs then realised that I can use on_click function in ui.scene() but don't know how to use it... Please tell me if anybody knows

**Comment by falkoschindler** [comment, marked as answer]
I'm afraid there is currently no way to add a button to a 3D scene. But using click events you should be able to implement a similar user interaction:

```py
from nicegui import events, ui

def handle_click(event: events.SceneClickEventArguments) -> None:
    ui.notify(f'{event.click_type} on {event.hits[0].object_name or event.hits[0].object_id}', flush=True)

with ui.scene(on_click=handle_click) as scene:
    scene.sphere().with_name('sphere')

ui.run()
```

The event object contains the click type (e.g. double click), mouse button, modifier keys, and a list of hits. This list is sorted by distance from the camera, so the first hit is the closest one. Each entry in the list contains the object ID, object name, and the point of intersection. See https://threejs.org/docs/#api/en/core/Raycaster for more information.
